src/linear_algebra/tensor_product.lean

```diff
@@ -805,6 +805,33 @@ rfl
   (congr f g).symm (p ⊗ₜ q) = f.symm p ⊗ₜ g.symm q :=
 rfl
 
+variables (R M N P Q)
+
+/-- This special case is worth defining explicitly since it is useful for defining multiplication
+on tensor products of modules carrying multiplications (e.g., associative rings, Lie rings, ...).
+
+E.g., suppose `M = P` and `N = Q` and that `M` and `N` carry bilinear multiplications:
+`M ⊗ M → M` and `N ⊗ N → N`. Using `map`, we can define `(M ⊗ M) ⊗ (N ⊗ N) → M ⊗ N` which, when
+combined with this definition, yields a bilinear multiplication on `M ⊗ N`:
+`(M ⊗ N) ⊗ (M ⊗ N) → M ⊗ N`. -/
+def comm_two_in_four : (M ⊗[R] N) ⊗[R] (P ⊗[R] Q) ≃ₗ[R] (M ⊗[R] P) ⊗[R] (N ⊗[R] Q) :=
+let e₁ := tensor_product.assoc R M N (P ⊗[R] Q),
+    e₂ := congr (1 : M ≃ₗ[R] M) (tensor_product.assoc R N P Q).symm,
+    e₃ := congr (1 : M ≃ₗ[R] M) (congr (tensor_product.comm R N P) (1 : Q ≃ₗ[R] Q)),
+    e₄ := congr (1 : M ≃ₗ[R] M) (tensor_product.assoc R P N Q),
+    e₅ := (tensor_product.assoc R M P (N ⊗[R] Q)).symm in
+(((e₁.trans e₂).trans e₃).trans e₄).trans e₅
+
+variables {M N P Q}
+
+@[simp] lemma comm_two_in_four_tmul (m : M) (n : N) (p : P) (q : Q) :
+  comm_two_in_four R M N P Q ((m ⊗ₜ n) ⊗ₜ (p ⊗ₜ q)) = (m ⊗ₜ p) ⊗ₜ (n ⊗ₜ q) :=
+rfl
+
+@[simp] lemma comm_two_in_four_symm_tmul (m : M) (n : N) (p : P) (q : Q) :
+  (comm_two_in_four R M N P Q).symm ((m ⊗ₜ p) ⊗ₜ (n ⊗ₜ q)) = (m ⊗ₜ n) ⊗ₜ (p ⊗ₜ q) :=
+rfl
```

Comment:
@fpvandoorn: I can't see to get `simps` to generate these - could you take a look at what's going on?

Reply:
`simps` only generates lemmas where the function is applied to a variable (or multiple variables), it doesn't know (yet?) what the structure of a type is. 
A simpler example: if you define `foo α ⊕ β ≃ β ⊕ α`, it can generate a lemma for `foo x`, but not for `foo (inl a)`.